src/wallet/wallet.h

```diff
@@ -205,21 +205,29 @@ class CAddressBookData
 {
 private:
     bool m_change{true};
+    bool m_used{false};
     std::string m_label;
+    std::map<std::string, std::string> m_receive_requests;
```

Comment:
I'm not convinced that this needs to be a map. Receive requests are only created by the GUI, and the GUI only creates them for a newly retrieved address, so it shouldn't be possible to have multiple receive requests.

Additionally, receive requests are identified by an int which is being (un)stringified. This could just be an int to avoid that constant conversion in the GUI.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/27224#discussion_r1137458690

> I'm not convinced that this needs to be a map. Receive requests are only created by the GUI, and the GUI only creates them for a newly retrieved address, so it shouldn't be possible to have multiple receive requests.
> 
> Additionally, receive requests are identified by an int which is being (un)stringified. This could just be an int to avoid that constant conversion in the GUI.

You could definitely be right about these things, but I need to look into GUI code more to have an opinion. This PR avoids changing any GUI code, and keeps the behavior of `interfaces::Wallet` methods the same as before from the perspective of the GUI.

If the definitions or behavior of `interfaces::Wallet` methods should change as you say, I think it would make sense to implement these changes as a separate PR, because making those changes here would not really simplify much (would basically replace `string` with `int` and `find()` with `has_value()`), and would require understanding GUI code to review, unlike the current changes.

Reply:
re: [#27224 (comment)](https://github.com/bitcoin/bitcoin/pull/27224#discussion_r1137458690)

Instead of extending the PR to modify GUI code, I've kept changes here internal to the wallet and added a note in `setAddressReceiveRequest` about the wallet/gui interface and say how it could be improved in the future.